yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -187,3 +195,90 @@ func (d *RangeTableFuncDetector) GetIssues() []QueryIssue {
 	}
 	return issues
 }
+
+type JsonConstructorFuncDetector struct {
+	query                                       string
+	unsupportedJsonConstructorFunctionsDetected mapset.Set[string]
+}
+
+func NewJsonConstructorFuncDetector(query string) *JsonConstructorFuncDetector {
+	return &JsonConstructorFuncDetector{
+		query: query,
+		unsupportedJsonConstructorFunctionsDetected: mapset.NewThreadUnsafeSet[string](),
+	}
+}
+
+func (j *JsonConstructorFuncDetector) Detect(msg protoreflect.Message) error {
```

Comment:
@sanyamsinghal fyi, here we seem to be dealing with multiple node types in a single detector. I personally don't see an issue with this approach. I think detectors can be defined in whichever way is convenient. 
The problem with sticking to one detector per node type(current standard) in this situation would be similar to the XML issues case : if you have multiple node types in the same query, parserIssueDetector will return multiple issues of same type for the query, which is not ideal. Writing merge logic is not ideal because then you'd have to merge even the details map field.

Let us know if you see any problem with such an approach.

Reply:
yeah makes sense now, lets have it flexible. What we just need to ensure is that we are not reporting the same issue for same query twice.